### PR TITLE
Remove useless traduction wordings and exception codes

### LINKED ISSUE #17969: Remove useless currency wording/exception
#### Is your feature request related to a problem?

The currency feature has evolved along the developments and some error messages are not useful any more:
`Cannot find a real currency matching this couple of ISO code and numeric ISO code`
Along with the exception code `CurrencyConstraintException::ISO_CODES_MISMATCH`

Because:
- for official currencies the numeric is code is now automatically filled
- for unofficial ones the numeric iso code is now null

#### Describe the solution you'd like

Remove all the code related to this removed feature:
- in controller for error messages
- in the exception for useless code
- in the behat context for unused statement
- DON'T FORGET to clean the default catalog as well and clean this useless message

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Remove useless traduction wordings and exception codes, they are no longer used due to refacto or specs evolution
| Type?         | refacto
| Category?     | LO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #17969
| How to test?  | No test needed, this is just cleaning

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/18652)

### CODE REVIEW

File: src/Core/Domain/Currency/Exception/CurrencyConstraintException.php
Comment (Progi1984) [inline src/Core/Domain/Currency/Exception/CurrencyConstraintException.php:79]: @jolelievre It's not a BC to remove a public constant ?

Comment (jolelievre) [reply src/Core/Domain/Currency/Exception/CurrencyConstraintException.php:79]: No because it was introduced in 177, so we can still clean it for now

Comment (Progi1984) [reply src/Core/Domain/Currency/Exception/CurrencyConstraintException.php:79]: Ok for me.

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by jolelievre** [comment]
Careful!! Better wait for @LouiseBonnard approval before merging this one!

**Comment by PierreRambaud** [comment]
Thanks @jolelievre

**Pull request merged**
